### PR TITLE
feat(proxy templates): reflect istio/istio templates additions

### PR DESCRIPTION
**Pull request by douglas-reid**
This PR adds the changes to the sidecar injection template that were added to istio/istio in PRs:
- https://github.com/istio/istio/pull/15551
- https://github.com/istio/istio/pull/16358

Full justification for changes can be found in those PRs.

### CODE REVIEW

**Comment by douglas-reid** [comment]
It appears that CI is broken for this repo?

**Comment by howardjohn** [comment]
We had some breakages and circle won't automatically rebase. Can you rebase this?

**Comment by kyessenov** [comment]
He's OoO. I'll take over before I myself go OoO once my PRs merge.

**Comment by douglas-reid** [comment]
so rebasing doesn't fix this issue.  what next?

**Comment by douglas-reid** [comment]
@howardjohn is the testing done with a version of istio from before the PRs that contain these changes?  I see a reference to a 8/14 image, but the changes in istio/istio happened after that.

**Review by costinm — APPROVED** [review]

**Comment by douglas-reid** [comment]
rebased again, in an attempt to get this in

**Comment by costinm** [comment]
What's the status here ?

**Comment by douglas-reid** [comment]
@costinm good question. i was never able to repro the failures, as I couldn't seem to get the tests to run for whatever reason.

this needs to be in... i'll just have to try again. open to suggestions.

File: istio-control/istio-autoinject/files/injection-template.yaml [hunk 1]
```diff
@@ -182,6 +182,10 @@ template: |
       valueFrom:
         fieldRef:
           fieldPath: status.podIP
+    - name: SERVICE_ACCOUNT
+        valueFrom:
```
Comment (morvencao) [inline istio-control/istio-autoinject/files/injection-template.yaml]: The indention is no right here.

Comment (douglas-reid) [reply istio-control/istio-autoinject/files/injection-template.yaml]: good catch!

**Pull request merged**